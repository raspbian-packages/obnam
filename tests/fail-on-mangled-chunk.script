#!/bin/sh
# Copyright 2011  Lars Wirzenius
# 
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

set -e

$SRCDIR/tests/backup

# Corrupt chunk files.
find "$DATADIR/repo/chunks" -type f -size +11c |
while read filename
do

    dd if="$filename" of="$filename.new" bs=1 count=10 2>/dev/null
    dd if="$filename" bs=1 skip=10 2>/dev/null | \
    tr '\0-\377' '\200-\377\0-\177' >> "$filename.new"
    mv "$filename.new" "$filename"
done

# Restore. This will fail, and output an error, which contains a chunk id.
# Blot out the id.
if $SRCDIR/tests/restore 2> "$DATADIR/stderr"
then
    cat "$DATADIR/stderr" 1>&2
    exit 0
else
    sed 's/chunk [[:digit:]]* checksum/chunk BLOTTED checksum/' \
        "$DATADIR/stderr" 1>&2
    exit 1
fi
