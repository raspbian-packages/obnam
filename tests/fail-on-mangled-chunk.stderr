ERROR: chunk BLOTTED checksum error
